### DISCUSSION TITLE
Updating Hymn libary to Hylang 0.27

### DISCUSSION THREAD

**Discussion by wrobell** [hylang/hy#2481] [Show and tell]
I have created a pull request for Hymn library to update it to Hylang 0.27: https://github.com/pyx/hymn/pull/25

Main focus of recent changes has been maybe monad. I will continue working on that pull request, and I would appreciate any feedback. Please do not hesitate to comment in the pull request.

**Comment by Kodiologist** [comment]
It looks like the last commit on that repository is from 2020. I wouldn't hold my breath on a response from the author. You can always develop your fork independently, though.

**Reply by wrobell** [reply]
I suspect so.

This is, at the moment, an exercise for me to learn few more things about Lisp and Hylang.

Time will show what will happen.

**Comment by pyx** [comment]
@wrobell Thank you for your contribution. I merged the PR yesterday.

The project is not dead, at least from my point of view.

Reasons of the lack of maintenance:
1. Hy had undergone so many changes last few years, removed `def` form, `defreader` to `deftag` back to `defreader`, removing `let` and putting it back, 0.20 to 1.0 then back to 0.24, to name but a few. It was hard for me to follow all the breaking changes as I have limited spare-time.
2. It seems no one use it besides myself and Tuula Turto (contributor of Hy as well), I knew because we had a few email exchanges talking about hymn being used in a game project back then. That game project was no longer updated and I have been using OCaml lately if I want to code something.

So, I decided that I will wait until when there is a more stable API set of Hy, then I will bring it up to date.

Now that we have at least one possible new user ;), it will be helpful to bring it up to date, PRs are welcome.
On top of changes that accommodate Hy's ever-changing APIs, I am thinking deeper changes, e.g. making use of `dataclass` and `match` statements, etc. It might not be worth it, though, because python doesn't support ADT (algebraic data types) as the ML family languages do.

**Reply by pyx** [reply]
Reference: related discussion https://github.com/hylang/hy/discussions/2473

**Comment by pyx** [comment]
Finally, I released ver 1.0 that support hy 0.27 today.
https://pypi.org/project/hymn/
